core/input/input_event.cpp

```diff
@@ -796,7 +868,40 @@ bool InputEventJoypadMotion::action_match(const Ref<InputEvent> &p_event, bool *
 	return match;
 }
 
+// Maps to 2*axis if value is neg, or + 1 if value is pos
+static const char *_joy_axis_descriptions[JOY_AXIS_MAX * 2] = {
+	TTRC("Left Stick Left, Joystick 0 Left"),
+	TTRC("Left Stick Right, Joystick 0 Right"),
+	TTRC("Left Stick Up, Joystick 0 Up"),
+	TTRC("Left Stick Down, Joystick 0 Down"),
+	TTRC("Right Stick Left, Joystick 1 Left"),
+	TTRC("Right Stick Right, Joystick 1 Right"),
+	TTRC("Right Stick Up, Joystick 1 Up"),
+	TTRC("Right Stick Down, Joystick 1 Down"),
+	TTRC("Joystick 2 Left"),
+	TTRC("Left Trigger, Sony L2, Xbox LT, Joystick 2 Right"),
+	TTRC("Joystick 2 Up"),
+	TTRC("Right Trigger, Sony R2, Xbox RT, Joystick 2 Down"),
+	TTRC("Joystick 3 Left"),
+	TTRC("Joystick 3 Right"),
+	TTRC("Joystick 3 Up"),
+	TTRC("Joystick 3 Down"),
+	TTRC("Joystick 4 Left"),
+	TTRC("Joystick 4 Right"),
+	TTRC("Joystick 4 Up"),
+	TTRC("Joystick 4 Down"),
+};
+
 String InputEventJoypadMotion::as_text() const {
+	String description = "";
+	if (axis < JOY_AXIS_MAX) {
+		description = vformat(" (%s)", RTR(_joy_axis_descriptions[2 * axis + (axis_value < 0 ? 0 : 1)]));
+	}
+
+	return vformat(RTR("Joypad Motion on Axis %s%s with Value %s"), itos(axis), description, String(Variant(axis_value)));
+}
```

Comment:
I think there is something wrong here. It looks like you're missing the `+` or `-` after the axis number and before the description in parentheses and instead are adding "with Value ±1" to the end. Is this intentional? This is what the Input Map Editor currently presents:

![Axis Selection](https://user-images.githubusercontent.com/9253928/99678818-237e0880-2a73-11eb-83ec-334b4c45e6ed.png)

Reply:
Yeah it looks wrong:

![image](https://user-images.githubusercontent.com/41730826/99960243-26ad2780-2dd8-11eb-8c95-4cd98db7a17b.png)

What about "Joypad Motion on Axis 0, Value -1 (Left Stick Left, Joystick 0 Left)", i.e. putting the value before the description. I think I actually meant to do this, as how I currently have it in the image above doesn't make much sense.

Reply:
Perhaps there does need to be a separation between the `as_text()` for the event and the 20 direction descriptions used by Input Map Editor.

Reply:
It would be nice if it could be avoided but maybe it can't. To keep the descriptions in one place though, a method to get the axis/button descriptions would need to be introduced (i.e. `get_axis_description(int axis)`)

Is there a particular issue you have with `Joypad Motion on Axis 0, Value -1 (Left Stick Left, Joystick 0 Left)`? This will also likely come along with a different event editor (shown below) which was a part of #42600 which I have now split up into multiple prs.

![new_dialog_demo](https://user-images.githubusercontent.com/41730826/99963449-8b1eb580-2ddd-11eb-8dc2-3bafdc718a66.gif)

Reply:
> Is there a particular issue you have with Joypad Motion on Axis 0, Value -1 (Left Stick Left, Joystick 0 Left)?

I just think there is a difference between the event description, which is axis specific and the Input Map description, which is half axis specific:
Event description: `Joypad Motion on Axis 0, (Left Stick X-Axis, Joystick 0 X-Axis) with Value -0.3`
Input Map description: `Joypad Axis 0 - (Left Stick Left, Joystick 0 Left)`

Reply:
Ok, I added a method called `get_axis_description()` so that you can get the axis description for a given `InputEventJoypadMotion`. This means the list of descriptions is still in once place, but the description for an event can still be accessed from anywhere. Thus, the text can still be constructed in `input_map_editor.cpp` but it is all on one line as opposed to previously many lines used to construct the description.

`vformat(TTR("Axis %s%s (%s)"), jm->get_axis(), jm->get_axis_value() < 0 ? "-" : "+", jm->get_axis_description());`

Reply:
I still think the differences between the event descriptions (10 axis specific), and the Input Map descriptions (20 half-axis specific) are enough to justify two different lists of descriptions for the two different needs:

**Event descriptions:**
- Joypad **Motion on** Axis 0, (Left Stick **X-Axis**, Joystick 0 **X-Axis**) **with Value -0.3**
- Joypad **Motion on** Axis 0, (Left Stick **X-Axis**, Joystick 0 **X-Axis**) **with Value 0.3**
- Joypad **Motion on** Axis 1, (Left Stick **Y-Axis**, Joystick 0 **Y-Axis**) **with Value -0.3**
- Joypad **Motion on** Axis 1, (Left Stick **Y-Axis**, Joystick 0 **Y-Axis**) **with Value 0.3**

**Input Map descriptions:**
- Joypad Axis 0 **–** (Left Stick **Left**, Joystick 0 **Left**)
- Joypad Axis 0 **+** (Left Stick **Right**, Joystick 0 **Right**)
- Joypad Axis 1 **–** (Left Stick **Up**, Joystick 0 **Up**)
- Joypad Axis 1 **+** (Left Stick **Down**, Joystick 0 **Down**)

I've highlighted the differences above for one joystick (two axes and the four directions).

Reply:
So do you propose a method specific for InputEventJoypadMotion? In addition to `as_text()`, something like `input_map_description()` (bad example but you know what I mean)?

Reply:
If it's specific to the input map, it should only be in the input map editor code IMO and not an extra method in InputEvents.

Reply:
Yep, makes sense

Reply:
Ok, all done. Changed `as_text()` to contain 10 options for InputEventJoyMotion, and added joy axis descriptions to Input Map Editor such that the description meets the above examples by @madmiraal.

See lines 358-390 of input_map_editor.cpp for the main changes in this iteration